Tests fail on darwin with the following log.
 `*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Critical error: required built-in appearance SystemAppearance not found'`

I suspect `CoreFoundation`, `AppKit`, or the matplotlib environment variable `MPLBACKEND`, but I can't fix it.
Maybe someone at @NixOS/darwin-maintainers  knows how to make it work.

<summary></summary><details>

```
kaleido-tests-aarch64-darwin> building '/nix/store/r6dib0z8rc7pbjhfqrh184a6vh9k9a8h-kaleido-tests.drv'
kaleido-tests-aarch64-darwin> Traceback (most recent call last):
kaleido-tests-aarch64-darwin>   File "/nix/store/i36nyn014clgrvrgmv60vg58wl0ifmvn-tests.py", line 11, in <module>
kaleido-tests-aarch64-darwin>     fig.write_image(outfile, engine="kaleido")
kaleido-tests-aarch64-darwin>   File "/nix/store/k4hlqc8dcdw3890fvghzl3sppwfcs78p-python3.12-plotly-5.23.0/lib/python3.12/site-packages/plotly/basedatatypes.py", line 3835, in write_image
kaleido-tests-aarch64-darwin>     return pio.write_image(self, *args, **kwargs)
kaleido-tests-aarch64-darwin>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kaleido-tests-aarch64-darwin>   File "/nix/store/k4hlqc8dcdw3890fvghzl3sppwfcs78p-python3.12-plotly-5.23.0/lib/python3.12/site-packages/plotly/io/_kaleido.py", line 266, in write_image
kaleido-tests-aarch64-darwin>     img_data = to_image(
kaleido-tests-aarch64-darwin>                ^^^^^^^^^
kaleido-tests-aarch64-darwin>   File "/nix/store/k4hlqc8dcdw3890fvghzl3sppwfcs78p-python3.12-plotly-5.23.0/lib/python3.12/site-packages/plotly/io/_kaleido.py", line 143, in to_image
kaleido-tests-aarch64-darwin>     img_bytes = scope.transform(
kaleido-tests-aarch64-darwin>                 ^^^^^^^^^^^^^^^^
kaleido-tests-aarch64-darwin>   File "/nix/store/y1fsgdcc12c2jyb1pal6ds0icrz7x8x2-python3.12-kaleido-0.2.1/lib/python3.12/site-packages/kaleido/scopes/plotly.py", line 153, in transform
kaleido-tests-aarch64-darwin>     response = self._perform_transform(
kaleido-tests-aarch64-darwin>                ^^^^^^^^^^^^^^^^^^^^^^^^
kaleido-tests-aarch64-darwin>   File "/nix/store/y1fsgdcc12c2jyb1pal6ds0icrz7x8x2-python3.12-kaleido-0.2.1/lib/python3.12/site-packages/kaleido/scopes/base.py", line 293, in _perform_transform
kaleido-tests-aarch64-darwin>     self._ensure_kaleido()
kaleido-tests-aarch64-darwin>   File "/nix/store/y1fsgdcc12c2jyb1pal6ds0icrz7x8x2-python3.12-kaleido-0.2.1/lib/python3.12/site-packages/kaleido/scopes/base.py", line 198, in _ensure_kaleido
kaleido-tests-aarch64-darwin>     raise ValueError(message)
kaleido-tests-aarch64-darwin> ValueError: Failed to start Kaleido subprocess. Error stream:
kaleido-tests-aarch64-darwin> 
kaleido-tests-aarch64-darwin> [0905/000022.334097:WARNING:resource_bundle.cc(431)] locale_file_path.empty() for locale en-US
kaleido-tests-aarch64-darwin> *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Critical error: required built-in appearance SystemAppearance not found'
kaleido-tests-aarch64-darwin> *** First throw call stack:
kaleido-tests-aarch64-darwin> (
kaleido-tests-aarch64-darwin> 	0   CoreFoundation                      0x0000000186e662ec __exceptionPreprocess + 176
kaleido-tests-aarch64-darwin> 	1   libobjc.A.dylib                     0x000000018694a158 objc_exception_throw + 60
kaleido-tests-aarch64-darwin> 	2   CoreFoundation                      0x0000000186e661dc +[NSException exceptionWithName:reason:userInfo:] + 0
kaleido-tests-aarch64-darwin> 	3   AppKit                              0x000000018a61e214 __33+[NSAppearance _initializeCoreUI]_block_invoke + 88
kaleido-tests-aarch64-darwin> 	4   libdispatch.dylib                   0x0000000186b613e8 _dispatch_client_callout + 20
kaleido-tests-aarch64-darwin> 	5   libdispatch.dylib                   0x0000000186b62c68 _dispatch_once_callout + 32
kaleido-tests-aarch64-darwin> 	6   AppKit                              0x000000018a63e5cc +[NSAppearance _aquaAppearance] + 60
kaleido-tests-aarch64-darwin> 	7   AppKit                              0x000000018a61d98c +[NSAppearance appearanceNamed:] + 32
kaleido-tests-aarch64-darwin> 	8   AppKit                              0x000000018a61d124 -[NSSystemAppearanceProxy init] + 124
kaleido-tests-aarch64-darwin> 	9   AppKit                              0x000000018a61d098 __38+[NSSystemAppearanceProxy systemProxy]_block_invoke + 24
kaleido-tests-aarch64-darwin> 	10  libdispatch.dylib                   0x0000000186b613e8 _dispatch_client_callout + 20
kaleido-tests-aarch64-darwin> 	11  libdispatch.dylib                   0x0000000186b62c68 _dispatch_once_callout + 32
kaleido-tests-aarch64-darwin> 	12  AppKit                              0x000000018a61d07c +[NSSystemAppearanceProxy systemProxy] + 64
kaleido-tests-aarch64-darwin> 	13  AppKit                              0x000000018a61d008 -[NSApplication(NSApplicationAppearance_Internal) _registerForAppearanceNotifications] + 32
kaleido-tests-aarch64-darwin> 	14  AppKit                              0x000000018a61acec -[NSApplication init] + 904
kaleido-tests-aarch64-darwin> 	15  AppKit                              0x000000018a61a798 +[NSApplication sharedApplication] + 128
kaleido-tests-aarch64-darwin> 	16  kaleido                             0x000000010908fb74 _ZN8headless27HeadlessContentMainDelegate24PreCreateMainMessageLoopEv + 36
kaleido-tests-aarch64-darwin> 	17  kaleido                             0x0000000106fb9748 _ZN7content21ContentMainRunnerImpl17RunServiceManagerERNS_18MainFunctionParamsEb + 188
kaleido-tests-aarch64-darwin> 	18  kaleido                             0x0000000106fb9660 _ZN7content21ContentMainRunnerImpl3RunEb + 304
kaleido-tests-aarch64-darwin> 	19  kaleido                             0x0000000106fb7e40 _ZN7content17RunContentProcessERKNS_17ContentMainParamsEPNS_17ContentMainRunnerE + 552
kaleido-tests-aarch64-darwin> 	20  kaleido                             0x0000000106fb8520 _ZN7content11ContentMainERKNS_17ContentMainParamsE + 48
kaleido-tests-aarch64-darwin> 	21  kaleido                             0x0000000107041240 _ZN8headless12_GLOBAL__N_114RunContentMainENS_15HeadlessBrowser7OptionsEN4base12OnceCallbackIFvPS1_EEE + 260
kaleido-tests-aarch64-darwin> 	22  kaleido                             0x0000000107041104 _ZN8headless19HeadlessBrowserMainENS_15HeadlessBrowser7OptionsEN4base12OnceCallbackIFvPS0_EEE + 56
kaleido-tests-aarch64-darwin> 	23  kaleido                             0x000000010444b8e4 main + 180
kaleido-tests-aarch64-darwin> 	24  dyld                                0x0000000186987154 start + 2476
kaleido-tests-aarch64-darwin> )
kaleido-tests-aarch64-darwin> /nix/store/y1fsgdcc12c2jyb1pal6ds0icrz7x8x2-python3.12-kaleido-0.2.1/lib/python3.12/site-packages/kaleido/executable/kaleido: line 5:  3017 Trace/BPT trap: 5       ./bin/kaleido $@
```
</details>

--- a/pkgs/development/python-modules/kaleido/default.nix
+++ b/pkgs/development/python-modules/kaleido/default.nix
@@ -112,5 +112,6 @@ buildPythonPackage rec {
     sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ]; # Trust me, I'm not happy. But after literal hours of trying to reverse-engineer their build system and getting nowhere, I'll use the stupid binaries >:(
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ pandapip1 ];
+    broken = stdenv.isDarwin; # Tests fail on darwin for some reason
   };
 }

